core/app/models/spree/calculator/weighted_rate.rb

```diff
@@ -0,0 +1,19 @@
+require_dependency 'spree/calculator'
+
+module Spree
+  class Calculator::WeightedRate < Calculator
+    preference :amount, :decimal, default: 0
+    preference :currency, :string, default: -> { Spree::Config[:currency] }
+
+    def compute_line_item(line_item)
+      if line_item && preferred_currency.casecmp(line_item.currency).zero?
```

Comment:
I liked having it in the calculator. I'm not sure it needs to exist in both places as it's the calculator that cares about this the most.